src/main/java/controller/GameController.java

```diff
@@ -1,82 +1,115 @@
 package controller;
 
-import domain.Command;
-import domain.DealerDto;
 import domain.ExceptionHandler;
-import domain.UserDto;
-import domain.deck.TotalDeck;
 import domain.deck.TotalDeckGenerator;
+import domain.game.Command;
 import domain.game.Game;
 import domain.game.State;
-import domain.user.Player;
-import domain.user.User;
-import domain.user.Users;
+import domain.money.Money;
+import domain.money.MoneyManager;
+import domain.money.Profit;
+import domain.user.*;
+import dto.UserDto;
 import view.InputView;
 import view.ResultView;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import static domain.game.State.BUST;
 import static domain.game.State.STAY;
 
 public class GameController {
 
     public void play() {
-        Users users = ExceptionHandler.handle(InputView::inputNames);
-        Game game = new Game(new TotalDeck(TotalDeckGenerator.generate()), users);
+        Users users = makeUsers();
+        Game game = new Game(TotalDeckGenerator.generate(), users);
         showStartStatus(users);
-
         hitOrStay(game, users);
-        showGameResult(users, game);
+        MoneyManager moneyManager = new MoneyManager(game.generatePlayerResults());
+        showResult(users, moneyManager);
+    }
+
+    private Users makeUsers() {
+        Names names = ExceptionHandler.handle(() -> new Names(getNames()));
+        List<Player> players = new ArrayList<>();
+        updatePlayers(names, players);
+        return new Users(Collections.unmodifiableList(players));
+    }
+
+    private List<Name> getNames() {
+        return ExceptionHandler.handle(() -> {
+            List<String> inputNames = InputView.inputNames();
+            return inputNames.stream()
+                    .map(Name::new)
+                    .collect(Collectors.toList());
+        });
+    }
+
+    private void updatePlayers(Names names, List<Player> players) {
+        for (Name name : names.value()) {
+            Money money = ExceptionHandler.handle(() -> new Money(InputView.inputBetting(name.value())));
+            Player player = new Player(name, money);
+            players.add(player);
+        }
```

Comment:
```suggestion
       private Users createUsers(Names names) {
          List<Player> players = new ArrayList<>();
          for (Name name : names.value()) {
              Money money = ExceptionHandler.handle(() -> new Money(InputView.inputBetting(name.value())));
              Player player = new Player(name, money);
              players.add(player);
          }
          return new Users(Collections.unmodifiableList(players));
    }
```
비어있는 players를 파라미터로 받아서 update 하는 로직이 어색한것 같네요
메서드 내에서 완전한 객체를 반환하도록 처리하면 좋을것 같네요

Reply:
감사합니다!